src/test/scala/DockerSuite.scala

```diff
@@ -0,0 +1,143 @@
+import com.typesafe.scalalogging.Logger
+import goatrodeo.omnibor.EdgeType
+import goatrodeo.omnibor.Item
+import goatrodeo.omnibor.ItemMetaData
+import goatrodeo.omnibor.Storage
+import goatrodeo.omnibor.ToProcess
+import goatrodeo.util.FileWrapper
+import org.json4s.JsonAST._
+import org.json4s._
+import org.json4s.native._
+
+import java.io.File
+import scala.collection.immutable.TreeMap
+import scala.collection.immutable.TreeSet
+
+class DockerSuite extends munit.FunSuite {
+  val logger = Logger(getClass())
+
+  test("Can build for a simple Docker file") {
+    val name = "test_data/download/docker_tests/bigtent_2025_03_22_docker.tar"
+
+    val nested = FileWrapper(File(name), name, None)
+    val store1 = ToProcess.buildGraphFromArtifactWrapper(nested)
+
+    val result = store1.purls()
+
+    assertEquals(
+      result,
+      TreeSet("pkg:docker/bigtent@2025_03_22")
+    )
+
+    val item = store1.read("pkg:docker/bigtent@2025_03_22").get
+    val aliasTo = item.connections
+      .collect { case (t, v) if EdgeType.isAliasTo(t) => v }
+      .headOption
+      .get
+
+    assertEquals(
+      aliasTo,
+      "gitoid:blob:sha256:7070a741d71c9e9e95c4b514a1fafc1b35275d512d9e95ea29fc2b075c03660f"
+    )
+
+    testLayersAndManifest(aliasTo, store1)
+
+  }
+
+  test("Merging names works") {
+    val a = ItemMetaData(TreeSet("foo"), TreeSet(), 1, TreeMap())
+    val b = ItemMetaData(TreeSet("bar"), TreeSet(), 1, TreeMap())
+
+    val aGitoids = () => Vector("yak", "moose")
+    val bGitoids = () => Vector("dog", "cat")
+
+    val mergedAA = a.merge(a, aGitoids, bGitoids)
+    assertEquals(a, mergedAA, "merging with self should be same")
+
+    val mergedAB = a.merge(b, aGitoids, bGitoids)
+    assertNotEquals(a, mergedAB, "They should differ")
+    assert(
+      mergedAB.fileNames.size == 6,
+      f"there should be 6 different filenames, but got ${mergedAB.fileNames}"
+    )
+  }
+
+  test("Can build for a complex file") {
+    val name = "test_data/download/docker_tests/grinder_bt_pg_docker.tar"
+
+    val nested = FileWrapper(File(name), name, None)
+    val store1 = ToProcess.buildGraphFromArtifactWrapper(nested)
+
+    val result = store1.purls()
+    val expectedpurls = TreeSet(
+      "pkg:docker/postgres@16.6",
+      "pkg:docker/postgres@9.6.12",
+      "pkg:docker/spicelabs%2Fbigtent@0.8.3",
+      "pkg:docker/spicelabs%2Fbigtent@latest",
+      "pkg:docker/spicelabs%2Fgrinder@0.1.0",
+      "pkg:docker/spicelabs%2Fgrinder@latest"
```

Comment:
Oof, %2F. ghcr.io purls are gonna be uuuugly

Reply:
Working on cleaning up the Docker pURLs